File: LeetCode-Java/Merge Two Sorted Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode tempNode = new ListNode(0);
        ListNode headNode = tempNode;
        while(l1 != null || l2 != null)
        {
            if(l1 == null)
            {
                headNode.next = l2;
                l2 = l2.next;
            }
            else if(l2 == null)
            {
                headNode.next = l1;
                l1 = l1.next;
            }
            else if(l1.val < l2.val)
            {
                headNode.next = l1;
                l1 = l1.next;
            }
            else
            {
                headNode.next = l2;
                l2 = l2.next;
            }
            headNode = headNode.next;
        }
        return tempNode.next;
    }
}

File: Leetcode-cpp/merge_two_sorted_lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode* fakeHead = new ListNode(0);
        ListNode* current = fakeHead;
        while (l1 != nullptr || l2 != nullptr) {
            if (l1 == nullptr || (l2 != nullptr && l1->val > l2->val)) {
                current->next = l2;
                l2 = l2->next;
                current = current->next;
            } else {
                current->next = l1;
                l1 = l1->next;
                current = current->next;
            }
        }
        return fakeHead->next;
    }
};